pkg/test/masteruserrecord/assertion.go

```diff
@@ -102,6 +105,20 @@ func (a *Assertion) HasStatusUserAccounts(targetClusters ...string) *Assertion {
 	return a
 }
 
+func (a *Assertion) HasExpectedProvisionedTime(expectedProvisionedTime metav1.Time) *Assertion {
+	err := a.loadUaAssertion()
+	require.NoError(a.t, err)
+
+	if expectedProvisionedTime == (metav1.Time{}) {
+		require.Nil(a.t, a.masterUserRecord.Status.ProvisionedTime)
+	} else {
+		// when the expected ProvisionedTime is not nil, check that the time was recently updated
+		require.WithinDuration(a.t, expectedProvisionedTime.Time, a.masterUserRecord.Status.ProvisionedTime.Time, 1*time.Second)
```

Comment:
btw, instead of this, you could expect a time after which the user was provisioned. 
In other words, in the `// given` part you would store the time `metav1.Now()`, then trigger the reconcile, and here you would verify that the actual provisioned time was set after the given time. With such a flow, you could easily debug the code, stop the execution for some time and the tests will still pass.

Reply:
That's a very good point about debugging the code and stopping the execution. It would be very confusing if the test fails if you wait for some time. Thanks for the suggestion!

Reply:
I agree with both Matous' comments. I think the name of the function is a bit misleading.
Can we rename it to `ProvisionedAfter(expectedProvisionedTime metav1.Time)` which will check that the provisioned time is set to something between the `expectedTIme` and `time.Now` (just to make sure it's not set to something in the future).